code/game/objects/structures/outfit_wardrobe.dm

```diff
@@ -0,0 +1,84 @@
+// if you need to access this elsewhere you already fucked up
+#define TRAIT_WARDROBE_USED "wardrobe_used"
+
+/obj/structure/outfit_wardrobe
+	name = "outfit wardrobe"
+	desc = "Peek in and select one of several snazzy outfits. Narnia not included."
+	icon = 'icons/obj/storage/closet.dmi'
+	icon_state = "fullcabinet"
+	obj_flags = INDESTRUCTIBLE
+	density = TRUE
+	anchored = TRUE
+	/**
+	 * Assoc list of outfits to how many charges left.
+	 * If value is null, the amount is infinite. Otherwise, it decreases by one per use.
+	 * At zero it can't be picked.
+	 * Value is unique per wardrobe.
+	 */
+	var/list/selectable_outfits_to_amount = list(
+		/datum/outfit/cat_butcher = null,
```

Comment:
Why not use `INFINITY` to denote infinite outfits?

Reply:
I thought that was frowned upon for some erason

Reply:
Its chill